sdks/go/test/integration/io/xlang/fileio/fileio_test.go

```diff
@@ -0,0 +1,80 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fileio
+
+import (
+	"encoding/json"
+	"flag"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	_ "github.com/apache/beam/sdks/v2/go/pkg/beam/runners/flink"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
+	"github.com/apache/beam/sdks/v2/go/test/integration"
+	"log"
+	"testing"
+)
+
+const (
+	labelFileIO = "fileio"
+)
+
+var expansionAddr string
+
+var flatSchemaElements = []interface{}{
+	&flatSchema{
+		AString:  "a",
+		AnInt32:  1,
+		AnInt64:  1001,
+		AFloat32: 1.1,
+		AFloat64: 1000.1,
+		ABoolean: true,
+	},
+}
+
+func checkFlags(t *testing.T) {
+	if *integration.StorageBucket == "" {
+		t.Skip("No Google Cloud Storage Bucket name provided.")
+	}
+}
+
+func TestExpandWrite(t *testing.T) {
+	integration.CheckFilters(t)
+	checkFlags(t)
+	write := WriteXLangPipeline(expansionAddr, "output", flatSchemaElements...)
+	read := ReadPipeline("output", "json", func(b []byte) (*flatSchema, error) {
+		var v *flatSchema
+		if err := json.Unmarshal(b, &v); err != nil {
+			return nil, err
+		}
+		return v, nil
+	})
+	ptest.RunAndValidate(t, write)
```

Comment:
@lostluck Good news. So far at least I can see Google Cloud storage objects with the elements written in JSON format. I just need to finish the `ptest.RunAndValidate(t, read)` for at least one test to pass.